packages/@vuepress/core/lib/node/loadConfig.js

```diff
@@ -28,9 +28,12 @@ module.exports = function loadConfig (vuepressDir, bustCache = true) {
   } else if (fs.existsSync(configTomlPath)) {
     siteConfig = parseConfig(configTomlPath)
   } else if (fs.existsSync(configPath)) {
-    siteConfig = require(configPath)
+    if (typeof require(configPath).then === 'function') {
+      siteConfig = require(configPath).then((config) => config)
```

Comment:
if you can add a catch with a pretty error message it will be good for me.

use logs in shared utils package to log the error

Reply:
I will

Reply:
Hi @f3ltron  not to much time to review and add a proper commit but updated with catch and logger.error message